test/suites/edgex-no-sec/ekuiper_test.go

```diff
@@ -64,4 +74,98 @@ func TestRulesEngine(t *testing.T) {
 
 		require.Greaterf(t, reading.TotalCount, 0, "No readings have been re-published to EdgeX message bus by ekuiper")
 	})
+
+}
+
+func subtestSetup(t *testing.T) (teardown func(), err error) {
+	start := time.Now()
+
+	teardown = func() {
+		log.Println("[SUBTEST TEARDOWN]")
+		utils.SnapDumpLogs(t, start, deviceVirtualSnap)
+		utils.SnapDumpLogs(t, start, ekuiperSnap)
+		utils.SnapDumpLogs(t, start, ascSnap)
+
+		log.Println("Removing installed snap:", !utils.SkipTeardownRemoval)
+		if !utils.SkipTeardownRemoval {
+			utils.SnapRemove(t, deviceVirtualSnap)
+			utils.SnapRemove(t, ekuiperSnap)
+			utils.SnapRemove(t, ascSnap)
+		}
+	}
+	defer teardown()
+
+	log.Println("[SUBTEST CLEAN]")
+	utils.SnapRemove(t, deviceVirtualSnap)
+	utils.SnapRemove(t, ekuiperSnap)
+	utils.SnapRemove(t, ascSnap)
+
+	log.Println("[SUBTEST SETUP]")
+
+	if err = utils.SnapInstallFromStore(t, deviceVirtualSnap, utils.ServiceChannel); err != nil {
+		teardown()
+		return
+	}
+
+	if err = utils.SnapInstallFromStore(t, ekuiperSnap, utils.ServiceChannel); err != nil {
+		teardown()
+		return
+	}
+
+	if err = utils.SnapInstallFromStore(t, ascSnap, utils.ServiceChannel); err != nil {
+		teardown()
+		return
+	}
+
+	// turn security off
+	utils.SnapSet(t, deviceVirtualSnap, "config.edgex-security-secret-store", "false")
+	utils.SnapSet(t, ascSnap, "config.edgex-security-secret-store", "false")
+	utils.SnapSet(t, ekuiperSnap, "config.edgex-security-secret-store", "false")
+
+	// use ASC for event filtering
+	utils.SnapSet(t, ekuiperSnap, "config.edgex.default.topic", "rules-events")
+	utils.SnapSet(t, ekuiperSnap, "config.edgex.default.messagetype", "event")
+	utils.SnapSet(t, ascSnap, "profile", "rules-engine")
+
+	// set tests to run without a config provider when testing config options as a temporary solution.
+	// update this once the following PR has been merged: https://github.com/canonical/edgex-snap-testing/pull/175
+	disableConfigProviderServiceSnap(t, deviceVirtualSnap, deviceVirtualApp)
+	disableConfigProviderServiceSnap(t, ascSnap, ascApp)
```

Comment:
The PR has been merged. But why do we even need to run these tests without using CP?

Reply:
To ensure that the eKuiper subtest runs successfully, it is necessary to set the following config options:
```
// turn security off
utils.SnapSet(t, deviceVirtualSnap, "config.edgex-security-secret-store", "false")
utils.SnapSet(t, ascSnap, "config.edgex-security-secret-store", "false")
utils.SnapSet(t, ekuiperSnap, "config.edgex-security-secret-store", "false")

// use ASC for event filtering
utils.SnapSet(t, ekuiperSnap, "config.edgex.default.topic", "rules-events")
utils.SnapSet(t, ekuiperSnap, "config.edgex.default.messagetype", "event")
```